docs/learn/documentation/versioned/jobs/configuration-table.html

```diff
@@ -822,6 +828,40 @@
                 </tr>
 
                 <tr>
+                    <th colspan="3" class="section" id="streams"><a href="../container/streams.html">Streams</a></th>
+                </tr>
+
+                <tr>
+                    <td class="property" id="streams-system">streams.<span class="stream">stream-id</span>.samza.system</td>
+                    <td class="default"></td>
+                    <td class="description">
+                        The <span class="system">system-name</span> of the system on which this stream will be accessed.
+                        This property binds the stream to one of the systems defined with the property
+                        systems.<span class="system">system-name</span>.samza.factory. <br>
+                        If this property isn't specified, it is inherited from job.default.system.
+                    </td>
+                </tr>
+
+                <tr>
+                    <td class="property" id="streams-physical-name">streams.<span class="stream">stream-id</span>.samza.physical.name</td>
```

Comment:
Hi, @navina and @vjagadish1989 the purpose of the logic stream-id has its extended meaning even in code. Currently, the stream name and system name are physical concepts that changes when the runtime environment changes (we have seen this when one user sending output to local.PageViewEventByMemberId fail to deploy to production since local cluster is not there and the input has to be queuing.PageViewEventByMemberId, even in the code). Using logic stream-id and make the physical system name and stream name a config value instead of part of the config variable name helps to make the code portable: in user code, they only need to refer to "PageViewEventByMemberId", which can be a Kafka topic in queuing by config, or a HDFS file by config. User won't need to re-compile there code to port over their output between different systems.

Co-existence of systems.&lt;xyz&gt;.streams.&lt;abc&gt;. and streams.&lt;abcId&gt;.system... is confusing. We might want to mark the whole section of systems.&lt;xyz&gt;.streams.&lt;abc&gt; as deprecated, and re-write w/ the new set of configuration to avoid this confusion.